src/buffer.c

```diff
@@ -309,19 +350,44 @@ int BUFFER_append(BUFFER_HANDLE handle1, BUFFER_HANDLE handle2)
         }
         else
         {
-            unsigned char* temp = (unsigned char*)realloc(b1->buffer, b1->size + b2->size);
-            if (temp == NULL)
+            if (b2->size ==0)
             {
-                /* Codes_SRS_BUFFER_07_023: [BUFFER_append shall return a nonzero upon any error that is encountered.] */
-                result = __LINE__;
+                // b2->size = 0, whatever b1->size is, do nothing
+                result = 0;
+            }
+            else if (b1->size == 0 && b2->size !=0)
+            {
+                unsigned char* temp = (unsigned char*)malloc(b2->size);
+                if (temp == NULL)
+                {
+                    /* Codes_SRS_BUFFER_07_023: [BUFFER_append shall return a nonzero upon any error that is encountered.] */
+                    result = __LINE__;
+                }
+                else
+                {
+                    b1->buffer = temp;
+                    memcpy(b1->buffer, b2->buffer, b2->size);
+                    b1->size = b2->size;
+                    result = 0;
+                }
             }
```

Comment:
when b1->size = 0, b1->buffer is "\0". so szie of realloc should be b1->size + b2->size -1. That's why we need to consider b1->size =0 and !=0 respectively.